testsuite/src/test/java/org/ironjacamar/common/metadata/common/ExpressionRestoreTestCase.java

```diff
@@ -0,0 +1,291 @@
+ * under the terms of the Eclipse Public License 1.0 as
+ * published by the Free Software Foundation.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the Eclipse
+ * Public License for more details.
+ *
+ * You should have received a copy of the Eclipse Public License 
+ * along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.ironjacamar.common.metadata.common;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for StringUtils.restoreExpression() methods
+ *
+ * @author <a href="vrastseluev@gmail.com">Vladimir Rastseluev</a>
+ *
+ */
+public class ExpressionRestoreTestCase
+{
+   /**
+    * expressions map
+    */
+   Map<String, String> map;
+
+   /**
+    * constant
+    */
+   private String key = "key";
+
+   /**
+    * Overwritten version of the test
+    * @param expected result
+    * @param key of the map
+    * @param v restored value
+    */
+   public void test(String expected, String key, String v)
+   {
+      test(expected, key, null, v);
+   }
+
+   /**
+    *
+    * Test the restoreExpression method with the expression map
+    * @param expected result
+    * @param key of the map
+    * @param subkey of the map
+    * @param v resolved value
+    */
+   public void test(String expected, String key, String subkey, String v)
+   {
+      assertEquals(expected, StringUtils.restoreExpression(map, key, subkey, v));
+   }
+
+   /**
+    * Basic initialization
+    */
+   @Before
+   public void setUp()
+   {
+      map = new HashMap<>();
+   }
+
+   /**
+    * Test complex key
+    */
+   @Test
+   public void keyTest()
+   {
+      map.put(key + "|subkey", "${value}");
+      test("${value}", key, "subkey", null);
+   }
+
+   /**
+    * Test complex key with expression
+    */
+   @Test
+   public void keyWithExpressionTest()
+   {
+      map.put(key + "|subkey", "${value}");
+      test("${value}", key, "${subkey}", null);
+   }
+
+   /**
+    * Test complex key with expression and a default value
+    */
+   @Test
+   public void keyWithExpressionAndValueTest()
+   {
+      map.put(key + "|subkey", "${value}");
+      test("${value}", key, "${subkey:subvalue}", null);
+   }
+
+   /**
+    * Test complex key with expression and a default value
+    */
+   @Test
+   public void keyWithIncorrectExpressionTest()
+   {
+      map.put(key + "|${subkey", "${value}");
+      test("${value}", key, "${subkey", null);
+   }
+
+   /**
+    * Test special case /
+    */
+   @Test
+   public void specialTest1()
+   {
+      map.put(key, "${/}");
+      test("${/}", key, "\\");
```

Comment:
here could be any symbol instead of "\\", test will pass anyway. The same is applicable for the next test too.